drivers/SmartThings/matter-switch/src/init.lua

```diff
@@ -723,40 +694,39 @@ local function detect_bridge(device)
 end
 
 local function device_init(driver, device)
-  if device.network_type == device_lib.NETWORK_TYPE_MATTER then
-    -- initialize_switch will create parent-child devices as needed for multi-switch devices.
-    -- However, we want to maintain support for existing MCD devices, so do not initialize
-    -- device if it has already been previously initialized as an MCD device.
-    -- Also, do not attempt a profile switch for a bridge device.
-    if not device:get_field(COMPONENT_TO_ENDPOINT_MAP) and
-       not device:get_field(SWITCH_INITIALIZED) and
-       not detect_bridge(device) then
-      -- create child devices as needed for multi-switch devices
-      initialize_switch(driver, device)
-    end
-    device:set_component_to_endpoint_fn(component_to_endpoint)
-    device:set_endpoint_to_component_fn(endpoint_to_component)
-    if device:get_field(IS_PARENT_CHILD_DEVICE) == true then
-      device:set_find_child(find_child)
-    end
-    local main_endpoint = find_default_endpoint(device)
-    for _, ep in ipairs(device.endpoints) do
-      if ep.endpoint_id ~= main_endpoint then
-        local id = 0
-        for _, dt in ipairs(ep.device_types) do
-          id = math.max(id, dt.device_type_id)
-        end
-        for _, attr in pairs(device_type_attribute_map[id] or {}) do
-          if id == GENERIC_SWITCH_ID and attr ~= clusters.PowerSource.attributes.BatPercentRemaining then
-            device:add_subscribed_event(attr)
-          else
-            device:add_subscribed_attribute(attr)
-          end
+  if device.network_type ~= device_lib.NETWORK_TYPE_MATTER then
+    return
+  end
+
+  local main_endpoint = find_default_endpoint(device)
+  if not device:get_field(COMPONENT_TO_ENDPOINT_MAP) and -- this field is only set for old MCD devices. See comments in the field def.
+     not device:get_field(SWITCH_INITIALIZED) and
+     not detect_bridge(device) then
+    -- initialize the main device card with buttons if applicable, and create child devices as needed for multi-switch devices.
+    initialize_buttons_and_switches(driver, device, main_endpoint)
+  end
+  device:set_component_to_endpoint_fn(component_to_endpoint)
+  device:set_endpoint_to_component_fn(endpoint_to_component)
+  if device:get_field(IS_PARENT_CHILD_DEVICE) then
+    device:set_find_child(find_child)
+  end
+  -- ensure subscription to all endpoint attributes- including those mapped to child devices
+  for _, ep in ipairs(device.endpoints) do
+    if ep.endpoint_id ~= main_endpoint then
+      local id = 0
+      for _, dt in ipairs(ep.device_types) do
+        id = math.max(id, dt.device_type_id)
+      end
+      for _, attr in pairs(device_type_attribute_map[id] or {}) do
+        if id == GENERIC_SWITCH_ID and attr ~= clusters.PowerSource.attributes.BatPercentRemaining then
```

Comment:
```suggestion
        if id == GENERIC_SWITCH_ID and attr ~= clusters.PowerSource.attributes.BatPercentRemaining 
           and attr ~= clusters.PowerSource.attributes.BatChargeLevel then
```
This should have been included in https://github.com/SmartThingsCommunity/SmartThingsEdgeDrivers/pull/1862, could you add it here?

Reply:
yeah no problem